### PR TITLE
Checking in Istio example

### PR DESCRIPTION
**Pull request by preddy727**
### What does this PR do?

Checking in Istio example with tetrate using EKS blueprint, adot and sending istio metrics to prometheus for grafana visualization 

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted. Consult the CONTRIBUTING guide for submitting pull-requests.

### More

- [x ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ x] Yes, I ran `pre-commit run -a` with this PR
- [ x] Yes, I have added a new example under [examples](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/examples) to support my PR (when applicable)
- [ x] Yes, I have updated the [Pages](https://github.com/aws-observability/terraform-aws-observability-accelerator/tree/main/docs) for this feature

**Note**: Not all the PRs required examples and docs.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: modules/eks-monitoring/patterns/istio/dashboards/istio-control-plane-dashboard.json
Comment (bonclay7) [inline modules/eks-monitoring/patterns/istio/dashboards/istio-control-plane-dashboard.json:1]: With Grafana operator managing the dashboards with GitOps, we moved all the dashboards [here](https://github.com/aws-observability/aws-observability-accelerator/tree/main/artifacts/grafana-dashboards/eks)
Please create a new directory for istio and add the dashboards there. Also, set the dashbaord `id` to `null`

File: modules/eks-monitoring/patterns/istio/main.tf [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+
+
+/*
+resource "grafana_dashboard" "istiomeshdashboard" {
+  count       = var.enable_dashboards ? 1 : 0
+  folder      = var.dashboards_folder_id
+  config_json = file("${path.module}/dashboards/istio-mesh-dashboard.json")
+}
+*/
```
Comment (bonclay7) [inline modules/eks-monitoring/patterns/istio/main.tf:200]: You can delete this if this is not needed anymore

File: modules/eks-monitoring/patterns/istio/main.tf [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  folder      = var.dashboards_folder_id
+  config_json = file("${path.module}/dashboards/istio-control-plane-dashboard.json")
+}
+
+resource "grafana_dashboard" "istioperformancedashboard" {
+  count       = var.enable_dashboards ? 1 : 0
+  folder      = var.dashboards_folder_id
+  config_json = file("${path.module}/dashboards/istio-performance-dashboard.json")
+}
+
```
Comment (bonclay7) [inline modules/eks-monitoring/patterns/istio/main.tf:219]: Move these to [Flux provisioning](https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/modules/eks-monitoring/patterns/java/main.tf#L35-L58) instead

File: modules/eks-monitoring/patterns/istio/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+output "grafana_dashboard_urls" {
```
Comment (bonclay7) [inline modules/eks-monitoring/patterns/istio/outputs.tf:1]: We won't have the outputs anymore :(

File: modules/eks-monitoring/patterns/istio/variables.tf [hunk 4]
```diff
@@ -0,0 +1,22 @@
+variable "enable_alerting_rules" {
```
Comment (bonclay7) [inline modules/eks-monitoring/patterns/istio/variables.tf:1]: Patterns have now have config objects 
- pattern variable https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/modules/eks-monitoring/patterns/java/variables.tf#L1-L26 and 
- eks module declaration https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/modules/eks-monitoring/variables.tf#L248-L270, customers can change these entirely
- we compute default values in locals https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/modules/eks-monitoring/locals.tf#L33-L55
- and we evaluate if customer provided something, else we use the defaults https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/modules/eks-monitoring/main.tf#L190

File: modules/eks-monitoring/patterns/istio/versions.tf [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      version = ">= 1.14"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = ">= 2.4.1"
+    }
+    grafana = {
+      source  = "grafana/grafana"
+      version = ">= 1.25.0"
+    }
```
Comment (bonclay7) [inline modules/eks-monitoring/patterns/istio/versions.tf:24]: Provider not needed anymore

**Review by bonclay7 — CHANGES_REQUESTED** [review]
Thanks a lot for this awesome, unfortunately this PR came in between a change in structure, so you will have to change a few stuff. Let me know if you need any help with this

**Comment by bonclay7** [comment]
@awsdabra @preddy727 let me know if you need any help!

File: examples/existing-cluster-istio-monitoring/README.md
[the remark this answers is not in the document]
Comment (preddy727) [reply examples/existing-cluster-istio-monitoring/README.md:1]: Renamed example folder to eks-istio.

File: examples/existing-cluster-istio-monitoring/README.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+The current example deploys the [AWS Distro for OpenTelemetry Operator](https://docs.aws.amazon.com/eks/latest/userguide/opentelemetry.html) for Amazon EKS with its requirements and make use of existing
+Amazon Managed Service for Prometheus and Amazon Managed Grafana workspaces.
+Tetrate Istio EKS blueprint add-on
+
+It provide an existing EKS cluster with an Istio add-on, OpenTelemetry collector,
+curated Grafana dashboards, Prometheus alerting and recording rules with multiple
+configuration options on the cluster infrastructure.
+
+
+## Prerequisites
```
[the remark this answers is not in the document]
Comment (preddy727) [reply examples/existing-cluster-istio-monitoring/README.md:15]: Moved detailed instructions to Github pages.   Added a page for Istio and also defined that new file.

File: examples/existing-cluster-istio-monitoring/README.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+Use the following command to create an Amazon Managed Service for Prometheus workspace:
+
+aws amp create-workspace \
+--alias $IAA_AMP_WORKSPACE_NAME \
+--region $IAA_AWS_REGION
+
+Creating the Amazon Managed service for Prometheus workspace takes just a few seconds.
+
+You can improve security and performance by creating VPC-endpoint for Amazon Managed Service for Prometheus. For more information, see Using Amazon Service for Prometheus with interface VPC endpoints.
```
[the remark this answers is not in the document]
Comment (preddy727) [reply examples/existing-cluster-istio-monitoring/README.md:37]: Took this step out.

File: examples/existing-cluster-istio-monitoring/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (preddy727) [reply examples/existing-cluster-istio-monitoring/README.md:15]: Updated github pages.

File: examples/existing-cluster-istio-monitoring/README.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+```
+7. [Grafana api key with Admin role](https://docs.aws.amazon.com/grafana/latest/userguide/API_key_console.html)
```
[the remark this answers is not in the document]
Comment (preddy727) [reply examples/existing-cluster-istio-monitoring/README.md:39]: Done.

File: examples/existing-cluster-istio-monitoring/README.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+2. Initialize terraform
+
+```console
+cd examples/existing-cluster-istio-monitoring
+terraform init
+```
+
+3. Update variables.tf. Uncomment each of the defaults below with actual values for the name of the EKS cluster, aws_region, Prometheus workspace id, grafana workspace id and grafana api key.
```
[the remark this answers is not in the document]
Comment (preddy727) [reply examples/existing-cluster-istio-monitoring/README.md:62]: Removed from readme and updated examples

**Comment by bonclay7** [comment]
@preddy727 I don't understand why you closed the PR

**Pull request closed without merging**